homeassistant/components/growatt_server/__init__.py

```diff
@@ -1 +1,25 @@
 """The Growatt server PV inverter sensor integration."""
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+
+async def async_setup(hass, config):
+    """Platform setup, do nothing."""
```

Comment:
If there's any chance this integration will support more than sensors in the future, we should move the central logic, config schema and device interaction to the component.

We also want to import the config yaml into a config entry.

Reply:
As far as i can see Growatt only sells PV inverters, so i don't think there will be anything other than sensors.

I tried the config yaml importing but i wasn't successfull.
The sensor gets added in yaml using
```
sensor:
  - platform: growatt_server
    username: USERNAME
    password: PASSWORD
```
i added
```
    if DOMAIN in config:
        for server in config[DOMAIN]:
            hass.async_create_task(
                hass.config_entries.flow.async_init(
                    DOMAIN,
                    context={"source": config_entries.SOURCE_IMPORT},
                    data=server,
                )
            )
```
To the above function
and have this extra function in my config_flow.py
```
    async def async_step_import(self, import_config):
        """Import a config entry from configuration.yaml."""
        if CONF_PLANT_ID not in import_config:
            login_response = self.api.login(
                import_config[CONF_USERNAME], import_config[CONF_PASSWORD]
            )
            user_id = login_response["userId"]
            plant_info = self.api.plant_list(user_id)

            import_config[CONF_PLANT_ID] = plant_info["data"][0]["plantId"]

        return self.async_create_entry(
            title=import_config[CONF_NAME], data=import_config
        )
```

Am i doing this the correct way?

Reply:
What about switches to turn the inverter on/off?

If we move to a component centric layout now we will be future proof forever.

Move the config section to under the integration domain. Then the above code should generally work. You will have to adjust the config schema though.

Reply:
I tried to add config validation for it but running into it claiming
`Invalid config for [growatt_server]: [username] is an invalid option for [growatt_server]. Check: growatt_server->username. (See ?, line ?).`
```
DEVICE_SCHEMA = vol.Schema(
    {
        vol.Optional(CONF_NAME): cv.string,
        vol.Optional(CONF_PLANT_ID): cv.string,
        vol.Required(CONF_USERNAME): cv.string,
        vol.Required(CONF_PASSWORD): cv.string,
    }
)

CONFIG_SCHEMA = vol.Schema(
    {DOMAIN: vol.All(cv.ensure_list, [DEVICE_SCHEMA])}, extra=vol.ALLOW_EXTRA
)
```
I took this from the arcam integration and changed it to fit my configuration. But i can't seem to get it to work.

Reply:
i'd want the configuration to be:
```
growatt_server:
  - username: USERNAME
    password: PASSWORD
```
Since people could have multiple accounts and they could have multiple plants.